### ISSUE TITLE
Some things...

### ISSUE THREAD

**Issue by Ghoost666** [wwmm/easyeffects#2393]
Hello

1. what is the latency on the bottom left corner showing? i tried to increase the quantum in pipewire but the latency in easyeffects never changes
1.1 and what is the latency on each app showing exactly? do the two latency add up?

2. i wanted to increase latency since i have some pops, but just if i use the pitchshifter plugin, there is also crackling the first few seconds if you enable the pitchshifter plugin

3. if you add a new band inside the EQ plugin then the filter type of all existing bands reset to "bell", this was also the case as i imported my settings into the flatpak version, all filters got imported as "bell" (i copied the profiles into the corresponding flatpak folder)

suggestion:
4. make a input field for each EQ band so we can precisly type in what number we need, the fineadjust got somehow reckt with the newest 7.0 version, its no longer possible to adjust in 0,1db steps and if you click long on a gain band then you never reach the number you actually want

**Comment by wwmm** [comment]
>what is the latency on the bottom left corner showing? i tried to increase the quantum in pipewire but the latency in easyeffects never changes

It is the extra latency introduced by the plugins in the corresponding EasyEffects pipeline. This is not the server latency. That is why changing the quantum has not direct effect on it in most of the situations.

>i wanted to increase latency since i have some pops, but just if i use the pitchshifter plugin, there is also crackling the first few seconds if you enable the pitchshifter plugin

Even with the new pitch shift plugin in our master branch? We moved from rubberband to soundtouch. So I was expecting the issues with the pitch shift to be gone.

>if you add a new band inside the EQ plugin then the filter type of all existing bands reset to "bell",

It is a known problem #1472 that is hard to fix in a way where everybody is happy.

> make a input field for each EQ band so we can precisly type in what number we need

This is discussed at #1383. It is a matter of figuring out a solution that does not look horrible.

**Comment by Ghoost666** [comment]
1. but the latency shown never changes even if i add/remove plugins in the pipeline (its always 32ms with 192khz samplerate)
2. i just tried 7.0.4 which still seems to use rubberband, but looking forward for 7.0.5 since i had also quality issues with rubberband, the sound demo of soundtech (on their website) seems promising
3. its not just the bands that got turned on being "bell" type ... but actually ALL bands reset to "bell"
4. hiding it in the menue of Q factor and the other options is probably the way to go, maybe even a "lock" feature so someone cant missclick on the bands anymore?

new:
5. i forgot this issue, if i change the type of the filter inside the eq plugin the menue gets "stuck" .. to remove it i have to click on the taskbar to be able to click on the next menue of the next band

**Comment by wwmm** [comment]
>but the latency shown never changes even if i add/remove plugins in the pipeline (its always 32ms with 192khz samplerate)

I think the value shown in the window is only updated when the pipeline restarts. This will trigger the signal that updates it. IT could be improved in the future. But I am not sure about how hard it would be.

>i just tried 7.0.4 which still seems to use rubberband, but looking forward for 7.0.5 since i had also quality issues with rubberband, the sound demo of soundtech (on their website) seems promising

Yes. It is not released yet.

>its not just the bands that got turned on being "bell" type ... but actually ALL bands reset to "bell"

Oh... This may be a regression. I do not remember this being the case in the past. But as we do not track each band state when recreating all of them (unfortunately adding/removing bands forces this) it may not be easy to avoid this without going back to the mode where we do not touch the band type at all. That is something we have to think about @Digitalone1.

>hiding it in the menue of Q factor and the other options is probably the way to go, maybe even a "lock" feature so someone cant missclick on the bands anymore?

missclick the bands?

> i forgot this issue, if i change the type of the filter inside the eq plugin the menue gets "stuck" .. to remove it i have to click on the taskbar to be able to click on the next menue of the next band

That is a gtk4 bug. It is taking forever for them to fix it..

**Comment by Ghoost666** [comment]
> missclick the bands?

well its not really nessecary but imo it would be nice to have a lock feature
what i mean is that its kinda easy to missclick on a band and the gain gets set to for example +30db, what you usually dont want
i personally would like to lock the graphic band adjustment, if we get a input field for gain inside the options menue

**Comment by wwmm** [comment]
Humm... INdividually locking each control would be annoying from an implementation point of view. Maybe trying to make the whole band container widget `insensitive` to input is an easy way to "lock the band".

**Comment by Ghoost666** [comment]
> Humm... INdividually locking each control would be annoying from an implementation point of view. Maybe trying to make the whole band container widget insensitive to input is an easy way to "lock the band".

no, locking each individual band is definitely unnessecary, just lock them all with one button :)

another issue related to 3.
since 7.0.4 i also got this strange issue:
6. if i load different profiles sometimes (not all the times) the profiles also gets loaded with each band set to "bell", reloading the profile then helps to actually get the right filter types (tho this might be just a display error, i havent tested for audibility difference)

**Comment by Digitalone1** [comment]
I do not remember all visible band types were reset to "Bell". Maybe this changed with the introduction of the listview?

**Comment by Digitalone1** [comment]
So I'm looking at this issue and it's very tricky to solve.

If the `num-bands` changes, all visible bands are set to **Bell**, all hidden bands to **Off**.

At the first sight this seems a correct behavior, but in reality it's wrong.

`num-bands` is not only triggered manually by the user, but also automatically by preset changes or APO import. 

This means that if a preset/APO format has a different number of bands than the current configuration with various filters different than Bell one, the visible bands are first set to what they should have, then they are reset to Bell.

This should be disabled and nothing should be done. But behaving that way we force the new additional bands to be Off by default and the user should reenable them manually which is somewhat cumbersome.

My suggestion to solve this is applying a GtkDropDown at the top of the band, just above the band setting button, to let the user quickly change the band type. But this dropdown might have different width based on filter labels and the space between bands might change.

So my solution is to use a fixed width dropdown with 3 characters tags not translatable for each type:

```
OFF --> Off
BEL --> Bell
HPA --> High Pass
HSH --> High Shelf
LPA --> Low Pass
LSH --> Low Shelf
NOT --> Notch
RES --> Resonance
APA --> All Pass
``` 

We explain then in the documentation and the dropdown in the band settings is retained for a more explanatory option.

This is my suggestion. Please, feel free to give a better hint, if you have one.

**Comment by Ghoost666** [comment]
> If the num-bands changes, all visible bands are set to Bell, all hidden bands to Off.

> At the first sight this seems a correct behavior, but in reality it's wrong.

> num-bands is not only triggered manually by the user, but also automatically by preset changes or APO import.

> This means that if a preset/APO format has a different number of bands than the current configuration with various filters different than Bell one, the visible bands are first set to what they should have, then they are reset to Bell.

> This should be disabled and nothing should be done. But behaving that way we force the new additional bands to be Off by default and the user should reenable them manually which is somewhat cumbersome.

Wait... im not sure i understand the problem here... why even reset current active filters to "bell" ?
if i understand this right the problem with not reseting them to bell is that added bands are actually "off" otherwise but imo added bands being off is way less of a problem then the whole resetting to "bell" mess... imo added bands maybe even "should" be off, specially if previous settings get loaded again, am i wrong or do i understand not the problem?

**Comment by Digitalone1** [comment]
> Wait... im not sure i understand the problem here... why even reset current active filters to "bell" ?

It's a convention we always had. Not that it's necessarily the right thing to do. I suppose it's because when you add a band, you want it enabled 99% of the cases. I didn't notice this issue because I use only the Bell filter.

> if i understand this right the problem with not reseting them to bell is that added bands are actually "off" otherwise but imo added bands being off is way less of a problem then the whole resetting to "bell" mess... imo added bands maybe even "should" be off, specially if previous settings get loaded again, am i wrong or do i understand not the problem?

I agree that they should be Off. Being Off solves also all those issues. The problem, from a user point of view, is that every time you add a band, you have to click on the band settings button, then change the type from the DropDown. Add also that GTK has this never ending problem of not automatically close the popup, even if you click outside, so you are forced to press ESC.

This is very cumbersome and not user friendly IMHO. That's why I think it's still a problem to be addresses, maybe with a DropDown on top of the band settings button the change to type quickly.

It would be so much easier if LSP had a band enable boolean option alongside the filter types from Bell to All Pass (excluding Off), but we have to deal with a single filters enum having Off included...

**Comment by Ghoost666** [comment]
> I agree that they should be Off. Being Off solves also all those issues. The problem, from a user point of view, is that every time you add a band, you have to click on the band settings button, then change the type from the DropDown. Add also that GTK has this never ending problem of not automatically close the popup, even if you click outside, so you are forced to press ESC.

personally i would definitely prefer the "new bands being off", specially since previous settings get loaded that might be totally unrelated to the current preset, that is potentially "loud noise" you can avoid

imo either reset the added bands completely and assign bell filter as standard type or have them off 

im not sure if this is possible but the easiest would be to just assign "bell" to the added filters and ignore already active bands, assigning "off" to the disabled bands should work in a similar way just the other way around or?

**Comment by Digitalone1** [comment]
> im not sure if this is possible but the easiest would be to just assign "bell" to the added filters and ignore already active bands, assigning "off" to the disabled bands should work in a similar way just the other way around or?

The problem is that when the number of bands changes, you don't know which are the "added filters". I mean, from a user point of view, you know it, but not in the code.

I'm okay to set Off as default.

**Comment by wwmm** [comment]
>My suggestion to solve this is applying a GtkDropDown at the top of the band, just above the band setting button, to let the user quickly change the band type.

>I'm okay to set Off as default.

I prefer to have them off by default instead of moving the band type button outside of the band menu. As we are not going to be able to please everybody anyone we should take the less complicated path.

**Comment by wwmm** [comment]
I have updated the master branch. Hidden bands are set to off and current bands type is left untouched when adding/reducing the number of equalizer bands.

**Comment by Digitalone1** [comment]
> I prefer to have them off by default instead of moving the band type button outside of the band menu. As we are not going to be able to please everybody anyone we should take the less complicated path.

It's okay. I will make a test anyway adding another DropDown on top of the band settings. 

I'm just curious to see how it will behave and how it looks. Will post the screenshots.

**Comment by wwmm** [comment]
>I'm just curious to see how it will behave and how it looks. Will post the screenshots.

No problems. But I imagine it will look weird. And it is a vertical space that would probably be better used on something like #1383 considering that many people seem to want a way to type the band gain.

**Comment by Digitalone1** [comment]
![Schermata del 2023-06-11 17-33-43](https://github.com/wwmm/easyeffects/assets/25790525/3c1bfaa8-052f-456a-addc-d206878d8b11)

![Schermata del 2023-06-11 17-34-09](https://github.com/wwmm/easyeffects/assets/25790525/d8d53bf6-30aa-4d8d-b030-a7fc52e69a30)

![Schermata del 2023-06-11 17-34-28](https://github.com/wwmm/easyeffects/assets/25790525/6d55ac94-bc54-4418-a844-8b87001d145d)

Used the tags mentioned in https://github.com/wwmm/easyeffects/issues/2393#issuecomment-1586082027

@wwmm What do you think?

**Comment by wwmm** [comment]
>What do you think?

I still think that if we are going to put more widgets in the vertical box we probably should do for #1383.  We can live with the band type dropdown only in the menu. An alternative way to input the gain seems to have more demand among the users.

**Comment by Digitalone1** [comment]
> I still think that if we are going to put more widgets in the vertical box we probably should do for #1383. We can live with the band type dropdown only in the menu. An alternative way to input the gain seems to have more demand among the users.

Okay. The problem with the editable band gain is that there's no suitable Gtk widget for it. The Spinbutton is too big. 
A Spinbutton without the buttons would be the ideal solution, but it's missing. Is it possible to remove the buttons from a Spinbutton?

**Comment by wwmm** [comment]
>Okay. The problem with the editable band gain is that there's no suitable Gtk widget for it. The Spinbutton is too big.

Yes. I still did not figure out an easy solution.

>A Spinbutton without the buttons would be the ideal solution, but it's missing. Is it possible to remove the buttons from a Spinbutton?

Not that I am aware of. The buttons are hardcoded in the spinbutton sources. We could use the functions that return the widget children and try to force them to be hidden but it is not a good idea. This approach relies on gtk developers not doing changes to how the buttons are being handled internally.

I have to take some time to see how hard it is to add locale based number validation to an GtkEntry. Unless there is some magic that can be done to the spinbuttons through css style classes a custom widget will probably be necessary to fix #1383.
